crates/ruff_python_parser/src/parser/mod.rs

```diff
@@ -785,33 +912,94 @@ impl RecoveryContextKind {
                         "The keyword is not allowed as a variable declaration name".to_string(),
                     )
                 } else {
-                    ParseErrorType::OtherError("Assignment target expected".to_string())
+                    ParseErrorType::OtherError("Expected an assignment target".to_string())
                 }
             }
             RecoveryContextKind::TypeParams => ParseErrorType::OtherError(
                 "Expected a type parameter or the end of the type parameter list".to_string(),
             ),
-            RecoveryContextKind::ImportNames => {
+            RecoveryContextKind::ImportFromAsNames => {
                 ParseErrorType::OtherError("Expected an import name or a ')'".to_string())
             }
+            RecoveryContextKind::ImportNames => {
```

Comment:
Same as above. The error messages will possibly be improved with new invalid program test cases.

Reply:
Makes sense. What is unclear to me is if we should have a dedicated variant for each possible syntax error in `ParseErrorType` instead of using `OtherError` for everything

Reply:
Yeah, I do plan to take a closer look at the error types in another PR.